### PR TITLE
CAP-33: Do not move reserves

### PR DESCRIPTION
**Pull request by jonjove**

### CODE REVIEW

File: core/cap-0033.md [hunk 1]
```diff
@@ -340,29 +339,29 @@ struct InnerTransactionResult
...
-When account `A` is-sponsoring-future-reserves-for account `B`, balance will be
-deducted from `A` and "stored" in any ledger entry/sub-entry for which `B` would
-need to provide reserve. Sponsoring account creation (if `B` does not yet exist)
-would deduct `2 * baseReserve` whereas sponsoring any other sub-entry would
-deduct just `baseReserve`. If `A` does not have sufficient available balance of
-native asset, then the operation fails.
+#### Sponsorship Bookkeeping
+When account `A` is-sponsoring-future-reserves-for account `B`, any reserve
+requirements that would normally accumulate on `B` will instead accumulate on
+`A` as reflected in `numSponsoring`. Both accounts are guaranteed to exist, but
```
Comment (leighmcculloch) [inline core/cap-0033.md:347]: I like the rename of `ed` to `ing`, it reads better and describes that it is an ongoing action which was less clear in the last revision.

File: core/cap-0033.md [hunk 2]
```diff
@@ -340,29 +339,29 @@ struct InnerTransactionResult
...
+this can fail. If such a change would cause `A` to pass below the reserve
+requirement, then the operation fails. The fact that these reserves are being
+provided by another account will be reflected on `B` in `numSponsored`.
+
+When a sponsored ledger entry or sub-entry is removed, `numSponsoring` is
+decreased on the sponsoring account and `numSponsored` is decreased on the
+sponsored account. Both accounts are guaranteed to exist, so this cannot fail.
```
Comment (leighmcculloch) [inline core/cap-0033.md:354]: Does the fact that both decrease mean that removing a sponsored signer automatically removes the sponsorship automatically mean those funds are immediately reclaimed by sponsor? That's pretty cool.

File: core/cap-0033.md [hunk 3]
```diff
@@ -467,51 +461,44 @@ Else if op.type() == UPDATE_SPONSORSHIP_SIGNER
...
-            Fail with UPDATE_SPONSORSHIP_UNDERFUNDED
+        If AvailableBalance(a, NATIVE) < requiredReserve
+            Fail with UPDATE_SPONSORSHIP_LOW_RESERVE
 
         If wasSponsored
-            acc.numSponsored -= mult
-        Set acc.ext.v(1)
-        Set acc.ext.v1().ext.v(0)
+            a.ext.v1().ext.v2().numSponsoring--
+        Clear a.ext.v1().ext.v2().signerSponsoringIDs[i].sponsoringID
```
Comment (leighmcculloch) [inline core/cap-0033.md:486]: It's not clear to me fro the changes in this pseudocode but will sponsor's funds used to cover the reserve of a sponsored account remain in the sponsors account? i.e. No funds will move, they just get locked up?

Comment (jonjove) [reply core/cap-0033.md:486]: Exactly, funds aren't moving at all anymore.

File: core/cap-0033.md [hunk 4]
```diff
@@ -340,29 +339,29 @@ struct InnerTransactionResult
...
-`(2 + numSubEntries - numSponsored) * baseReserve + liabilities.selling`.
+`(2 + numSubEntries + numSponsoring - numSponsored) * baseReserve + liabilities.selling`.
+
+#### Sponsorship Bookkeeping
+When account `A` is-sponsoring-future-reserves-for account `B`, any reserve
+requirements that would normally accumulate on `B` will instead accumulate on
+`A` as reflected in `numSponsoring`. Both accounts are guaranteed to exist, but
+this can fail. If such a change would cause `A` to pass below the reserve
+requirement, then the operation fails. The fact that these reserves are being
+provided by another account will be reflected on `B` in `numSponsored`.
```
Comment (ire-and-curses) [inline core/cap-0033.md:350]: I really like that you can see the relationship from either A or B, nice

**Comment by tomquisel** [comment]
This change looks great to me!

**Pull request merged**